go-controller/pkg/config/config.go

```diff
@@ -1637,3 +1671,36 @@ func UpdateOVNNodeAuth(masterIP []string, southboundDBPort, northboundDBPort str
 	OvnNorth.updateIP(masterIP, northboundDBPort)
 	OvnSouth.updateIP(masterIP, southboundDBPort)
 }
+
+// buildOvnKubeNodeConfig updates OvnKubeNode config from cli and config file
+func buildOvnKubeNodeConfig(ctx *cli.Context, cli, file *config) error {
+	// Copy config file values over default values
+	if err := overrideFields(&OvnKubeNode, &file.OvnKubeNode, &savedOvnKubeNode); err != nil {
+		return err
+	}
+	//cli.OvnKubeNode.Mode = ctx.String("ovnkube-node-mode")
+
+	// And CLI overrides over config file and default values
+	if err := overrideFields(&OvnKubeNode, &cli.OvnKubeNode, &savedOvnKubeNode); err != nil {
+		return err
+	}
+
+	// validate ovnkube-node-mode
```

Comment:
can we warp this validation in a function?

Reply:
Sure